civis/io/_files.py

```diff
@@ -89,52 +223,43 @@ def file_to_civis(buf, name, api_key=None, client=None, **kwargs):
     pass to this function, do so using the ``'rb'`` (read binary)
     mode (e.g., ``open('myfile.zip', 'rb')``).
 
-    If you have the `requests-toolbelt` package installed
-    (`pip install requests-toolbelt`) and the file-like object is seekable,
-    then this function will stream from the open file pointer into Platform.
-    If `requests-toolbelt` is not installed or the file-like object is not
-    seekable, then it will need to read the entire buffer into memory before
-    writing.
+    Warning: If the file-like object is seekable, the current
+    position will be reset to 0.
+
+    This facilitates retries and is used to chunk files for multipart
+    uploads for improved performance.
+
+    Small or non-seekable file-like objects will be uploaded with a
+    single post.
     """
     if client is None:
         client = APIClient(api_key=api_key)
-    file_response = client.files.post(name, **kwargs)
 
-    # Platform has given us a URL to which we can upload a file.
-    # The file must be uploaded with a POST formatted as per
-    # http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
-    # Note that the payload must have "key" first and "file" last.
-    form = file_response.upload_fields
-    form_key = OrderedDict(key=form.pop('key'))
-    form_key.update(form)
-    form_key['file'] = buf
+    file_size = _buf_len(buf)
+    if not file_size:
+        log.warning('Could not determine file size; defaulting to '
+                    'single post. Files over 5GB will fail.')
```

Comment:
Will the typical use case

```python
with open("my_file.csv", "r") as f:
    file_to_civis(f, "my_file")
```

throw a warning?

Reply:
it should not. this would only happen in rare cases where someone was using a file-like object that does not have a len attribute or support os.fstat.

An example of one scenario is in the test_io in the test_zip_member_to_civis test function, where someone opened a zip file and opened an archive member in memory instead of extracted it to disk. The zipfile class implementation does not support os.fstat on archive members.